RevenueCatUI/CustomerCenter/ViewModels/BaseManageSubscriptionViewModel/BaseManageSubscriptionViewModel.swift

```diff
@@ -0,0 +1,337 @@
+            self.purchasesProvider = purchasesProvider
+            self.refundRequestStatus = refundRequestStatus
+            self.actionWrapper = actionWrapper
+            self.loadPromotionalOfferUseCase = loadPromotionalOfferUseCase
+            ?? LoadPromotionalOfferUseCase(purchasesProvider: purchasesProvider)
+            self.restoreAlertType = .loading
+        }
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    func determineFlow(for path: CustomerCenterConfigData.HelpPath, activeProductId: String? = nil) async {
+        // Convert the path to an appropriate action using the extension
+        if let action = path.asAction() {
+            // Send the action through the action wrapper
+            self.actionWrapper.handleAction(.buttonTapped(action: action))
+        }
+
+        switch path.detail {
+        case let .feedbackSurvey(feedbackSurvey):
+            self.feedbackSurveyData = FeedbackSurveyData(
+                configuration: feedbackSurvey,
+                path: path) { [weak self] in
+                    Task {
+                        await self?.onPathSelected(path: path)
+                    }
+                }
+
+        case let .promotionalOffer(promotionalOffer) where purchaseInformation?.store == .appStore:
+            if promotionalOffer.eligible {
+                self.loadingPath = path
+                let result = await loadPromotionalOfferUseCase.execute(promoOfferDetails: promotionalOffer)
+                switch result {
+                case .success(let promotionalOfferData):
+                    self.promotionalOfferData = promotionalOfferData
+                case .failure:
+                    await self.onPathSelected(path: path)
+                    self.loadingPath = nil
+                }
+            } else {
+                Logger.debug(Strings.promo_offer_not_eligible_for_product(
+                    promotionalOffer.iosOfferId, activeProductId ?? ""
+                ))
+                await self.onPathSelected(path: path)
+            }
+
+        default:
+            await self.onPathSelected(path: path)
+        }
+    }
+
+    func onDismissInAppBrowser() {
+        self.inAppBrowserURL = nil
+    }
+
+    func openManagementURL(for productId: String) {
+        guard let purchaseInformation,
+              purchaseInformation.productIdentifier == productId,
+        let url = purchaseInformation.managePurchaseURL else {
+            return
+        }
+
+        self.inAppBrowserURL = IdentifiableURL(url: url)
+    }
+
+#endif
+
+}
+
+// MARK: - Promotional Offer Sheet Dismissal Handling
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+extension BaseManageSubscriptionViewModel {
+
+    /// Function responsible for handling the user's action on the PromotionalOfferView
+    func handleDismissPromotionalOfferView(_ userAction: PromotionalOfferViewAction) async {
```

Comment:
yeah, related to my comment above, this feels like more appropriate naming for action handling